src/plugin.rs

```diff
@@ -1,14 +1,54 @@
+use crate::PlaySoundEvent;
 use bevy::app::{App, Plugin};
+use bevy::audio::AudioSinkPlayback;
 use bevy::log::{debug, trace};
-use bevy::prelude::{EventReader, NonSend, PostUpdate, Resource, Startup, Update, World};
+use bevy::math::Vec3;
+use bevy::prelude::{
+    Added, Commands, Component, Entity, EventReader, GlobalTransform, NonSend, PostUpdate, Query,
+    Res, Resource, Startup, Update, World,
+};
+use bevy::time::Time;
 use bevy_mod_sysfail::sysfail;
 use libfmod::ffi::{FMOD_INIT_NORMAL, FMOD_STUDIO_INIT_NORMAL, FMOD_STUDIO_LOAD_BANK_NORMAL};
-use libfmod::{EventDescription, Studio};
-use play_sound_event::PlaySoundEvent;
+use libfmod::{Attributes3d, EventDescription, Studio, Vector};
 use std::env::var;
 use std::fs::{canonicalize, read_dir};
 use std::path::{Path, PathBuf};
 
+#[derive(Component)]
+pub struct FmodListener {
+    previous_position: Vec3,
+}
+
+impl Default for FmodListener {
+    fn default() -> Self {
+        FmodListener {
+            previous_position: Vec3::default(),
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct FmodAudioSource {
+    pub name: &'static str,
+}
```

Comment:
This is basically `PlaySoundEvent`. We should decide on one approach before merging this.

Reply:
Latest commit contains my take on `PlaySoundEvent` which doesn't use Bevy events. However, it's still a different approach to what I did for spatial audio. I think if we go for one approach either the minimal example will become a bit more complicated (i.e. spawn an entity which plays an event once and then despawns itself) or my implementation for spatial will become a lot different and further removed from what I think is idiomatic. Let me know what you think

Reply:
Btw I'm not saying my take on `PlaySoundEvent` is the way to go, just spitballing